lib/src/typings/overrides.ts

```diff
@@ -39,34 +17,29 @@ type Classes<T> = Partial<
 >;
 
 export interface MuiPickersOverrides {
-  MuiPickersDay?: Classes<typeof DayStyles>;
-  MuiPickerDTHeader?: Classes<typeof DTHeaderStyles>;
-  MuiPickerDTTabs?: Classes<typeof DTTabsStyles>;
-  MuiPickersCalendar?: Classes<typeof CalendarStyles>;
-  MuiPickersCalendarView?: Classes<typeof CalendarViewStyles>;
-  MuiPickersCalendarHeader?: Classes<typeof CalendarHeaderStyles>;
-  MuiPickersSlideTransition?: Classes<typeof SlideTransitionStyles>;
-  MuiPickersYearSelection?: Classes<typeof MuiPickersYearSelectionStyles>;
-  MuiPickersYear?: Classes<typeof MuiPickersYearStyles>;
-  MuiPickersMonthSelection?: Classes<typeof MuiPickersMonthSelectionStyles>;
-  MuiPickersMonth?: Classes<typeof MuiPickersMonthStyles>;
-  MuiPickersTimePickerToolbar?: Classes<typeof TimePickerToolbarStyles>;
-  MuiPickersClock?: Classes<typeof ClockStyles>;
-  MuiPickersClockNumber?: Classes<typeof ClockNumberStyles>;
-  MuiPickersClockPointer?: Classes<typeof ClockPointerStyles>;
-  MuiPickersModal?: Classes<typeof ModalDialogStyles>;
-  MuiPickersToolbar?: Classes<typeof PickerToolbarStyles>;
-  MuiPickersToolbarButton?: Classes<typeof ToolbarButtonStyles>;
-  MuiPickersToolbarText?: Classes<typeof MuiPickersToolbarTextStyles>;
-  MuiPickersDatePickerRoot?: Classes<typeof DatePickerRootStyles>;
-  MuiPickerDTToolbar?: Classes<typeof MuiPickerDTToolbarStyles>;
-  MuiBasePickerStyles?: Classes<typeof MuiBasePickerStyles>;
-  // consider using inline import type notation
-  MuiPickersDesktopDateRangeCalendar?: Classes<
-    typeof import('../DateRangePicker/DateRangePickerViewDesktop').useStyles
-  >;
   MuiPickersArrowSwitcher?: Classes<typeof import('../_shared/ArrowSwitcher').useStyles>;
-  MuiPickersDateRangePickerInput?: Classes<
-    typeof import('../DateRangePicker/DateRangePickerInput').useStyles
-  >;
+  MuiPickersBasePickerStyles?: Classes<typeof import ('../Picker/Picker').useStyles>;
```

Comment:
Most of the exported override keys here seem to be from private components, so to be removed.
Also, in https://github.com/mui-org/material-ui/issues/19427, it seems that we make the DX simpler for TypeScript users.

Reply:
So the idea there is to stop having this central "registry" in overrides.ts and instead only declaring the export in the respective component's file?
(Something like that would make it easier to keep privateness in sync.)

Reply:
What makes a component public (or private)? For instance, if I don't see the `ArrowSwitcher` component exported from `src/index.ts`, does that mean that it's private? And also, assuming that was right, should I remove it from `overrides.ts` and call its styles `PrivateArrowSwitcher` instead?

Will we also have a `buildStyleName()` variant or parameter for private components?

Reply:
> What makes a component public (or private)?

Great question. So far on https://github.com/mui-org/material-ui it has been as simple as: Is the module exported from the barrel index? If exported public, if not, private.

Reply:
Side question: Is it intended that things like `defaultReduceAnimations` are exported from the top-level `index.ts`?

Reply:
Do the `overrides.ts` exports and the `index.ts` exports necessarily have to be in sync? Is that a best practice? What are the advantages?

Anyway I pushed a commit that hopefully does that.

Reply:
So far, we have been keeping them in sync. So unless there is a good reason to deviate from this rule, I would be in favor of sticking to it.